ml-proto/host/builtins.ml

```diff
@@ -1,9 +1,50 @@
 open Source
+open Eval
+open Memory
+open Types
```

Comment:
A good rule is to limit `open` to a small number of imports that are heavily used, e.g. those that define datatypes with many constructors (like Ast), but is best avoided in most other cases, because it quickly harms readability. (In general, I feel that we have already started overusing it.)